ci/main.go

```diff
@@ -2,31 +2,108 @@ package main
 
 import (
 	"context"
+	"fmt"
 	"log/slog"
 	"os"
 
-	"dagger.io/dagger"
+	"container-registry.com/harbor-satellite/ci/internal/dagger"
 )
 
-func main() {
-	ctx := context.Background()
+const (
+	DEFAULT_GO = "golang:1.22"
+	PROJ_MOUNT = "/app"
+	OUT_DIR    = "/binaries"
+	DOCKER_PORT =- 2375
+)
 
-	// initialize Dagger client
-	client, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))
-	if err != nil {
-		panic(err)
+type HarborSatellite struct{}
+
+func (m *HarborSatellite) Start(ctx context.Context, name string, source *dagger.Directory, release *dagger.Directory, GITHUB_TOKEN, VERSION, REPO_OWNER, REPO_NAME, RELEASE_NAME string) {
+
+	if name == "" {
+		slog.Error("Please provide the app name (satellite or ground-control) as an argument")
+		os.Exit(1)
 	}
-	defer client.Close()
 
-	// use a golang:1.19 container
-	// get version
-	// execute
-	golang := client.Container().From("golang:1.21").WithExec([]string{"go", "version"})
+	switch name {
+	case "satellite":
+		slog.Info("Starting satellite CI")
+		err := m.StartSatelliteCi(ctx, source, release, GITHUB_TOKEN, VERSION, REPO_OWNER, REPO_NAME, RELEASE_NAME, name)
+		if err != nil {
+			slog.Error("Failed to start satellite CI")
+			os.Exit(1)
+		}
+	case "ground-control":
+		slog.Info("Starting ground-control CI")
+		err := m.StartGroundControlCI(ctx, source, release, GITHUB_TOKEN, VERSION, REPO_OWNER, REPO_NAME, RELEASE_NAME, name)
+		if err != nil {
+			slog.Error("Failed to complete ground-control CI")
+			os.Exit(1)
+		}
+	default:
+		slog.Error("Invalid app name. Please provide either 'satellite' or 'ground-control'")
+		os.Exit(1)
+	}
+}
```

Comment:
Find a better way to implement...

### Suggestions:
- can implement functions such as `buildsatellite`.
- dagger should more likely to be used as a lego instead of a single workflow.
- so build and release should be a separate functions.